llvm/lib/ObjCopy/ELF/ELFObject.cpp

```diff
@@ -2206,8 +2206,15 @@ Error Object::removeSections(
   // Transfer removed sections into the Object RemovedSections container for use
   // later.
   std::move(Iter, Sections.end(), std::back_inserter(RemovedSections));
-  // Now finally get rid of them all together.
+  // Now get rid of them altogether.
   Sections.erase(Iter, std::end(Sections));
+
+  // Finally iterate over all sections and erase empty SHT_GROUP sections.
+  for (auto Iter = Sections.begin(); Iter != Sections.end(); ++Iter)
+    if (auto GroupSec = dyn_cast<GroupSection>(Iter->get()))
+      if (GroupSec->getMembersCount() == 0)
+        Sections.erase(Iter);
```

Comment:
This is UB and has a quadratic time complexity if the erased elements are many. https://eel.is/c++draft/vector#modifiers-3

"Effects: Invalidates iterators and references at or after the point of the erase."

You could utilize the return address of erase, but to address the time complexity issue you'd need `erase_if` anyway.

Reply:
Fixed, thanks!